controllers/datadogagent/feature/dogstatsd/feature.go

```diff
@@ -0,0 +1,198 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package dogstatsd
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	"sigs.k8s.io/yaml"
+
+	apicommon "github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	apicommonv1 "github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/volume"
+)
+
+func init() {
+	err := feature.Register(feature.DogStatsDIDType, buildDogStatsDFeature)
```

Comment:
```suggestion
	err := feature.Register(feature.DogstatsdIDType, buildDogstatsdFeature)
```

sorry for the pain, do you mind changing the casing to be consistent with the [feature name](https://github.com/DataDog/datadog-operator/blob/main/apis/datadoghq/v2alpha1/datadogagent_types.go#L68)? we eventually settled on this with agent core.